src/peer-store/persistent/index.js

```diff
@@ -0,0 +1,225 @@
+'use strict'
+
+const debug = require('debug')
+const log = debug('libp2p:persistent-peer-store')
+log.error = debug('libp2p:persistent-peer-store:error')
+
+const { Key } = require('interface-datastore')
+const multiaddr = require('multiaddr')
+const PeerId = require('peer-id')
+
+const PeerStore = require('..')
+
+const {
+  ADDRESS_NAMESPACE,
+  COMMON_NAMESPACE,
+  PROTOCOL_NAMESPACE
+} = require('./consts')
+
+const Addresses = require('./pb/address-book.proto')
+const Protocols = require('./pb/proto-book.proto')
+
+/**
+ * Responsible for managing the persistence of data in the PeerStore.
+ */
+class PersistentPeerStore extends PeerStore {
+  /**
+   * @constructor
+   * @param {Object} properties
+   * @param {Datastore} properties.datastore Datastore to persist data.
+   * @param {number} [properties.threshold = 5] Number of dirty peers allowed before commit data.
+   */
+  constructor ({ datastore, threshold = 5 }) {
+    super()
+
+    /**
+     * Backend datastore used to persist data.
+     */
+    this._datastore = datastore
+
+    /**
+     * Peers modified after the latest data persisted.
+     */
+    this._dirtyPeers = new Set()
+
+    this.threshold = threshold
+    this._addDirtyPeer = this._addDirtyPeer.bind(this)
+  }
+
+  /**
+   * Start Persistent PeerStore.
+   * @return {Promise<void>}
+   */
+  async start () {
+    log('Persistent PeerStore is starting')
```

Comment:
Since the debug namespace is `libp2p:persistent-peer-store` I think you can drop `Persistent` in these logs.